CodePush.ios.js

```diff
@@ -55,45 +51,64 @@ var getSdk = (() => {
   }
 })();
 
+function getCurrentPackage() {
+  return new Promise((resolve, reject) => {
+    var localPackage;
+    NativeCodePush.getCurrentPackage()
+      .then((currentPackage) => {
+        localPackage = currentPackage;
+        return NativeCodePush.isFailedUpdate(currentPackage.packageHash);
+      })
+      .then((failedUpdate) => {
+        localPackage.failedApply = failedUpdate;
+        resolve(localPackage);
+      })
+      .catch(reject)
+      .done();
+  });
+}
+
 function checkForUpdate() {
   var config;
   var sdk;
+  
   return getConfiguration()
-    .then((configResult) => {
-      config = configResult;
-      return getSdk();
-    })
-    .then((sdkResult) => {
-      sdk = sdkResult;
-      return getCurrentPackage();
-    })
-    .then((localPackage) => {
-      var queryPackage = {appVersion: config.appVersion};
-      if (localPackage && localPackage.appVersion === config.appVersion) {
-        queryPackage = localPackage;
-      }
-
-      return new Promise((resolve, reject) => {
-        sdk.queryUpdateWithCurrentPackage(queryPackage, (err, update) => {
-          if (err) return reject(err);
-          if (update) {
-            // There is an update available for a different native app version. In the current version of this plugin, we treat that as no update.
-            if (update.updateAppVersion) resolve(false);
```

Comment:
It looks like the deletion of this code is a regression. https://github.com/Microsoft/code-push/issues/50